executor/compiler.go

```diff
@@ -157,6 +157,11 @@ func (c *Compiler) Compile(ctx context.Context, stmtNode ast.StmtNode) (_ *ExecS
 			}
 		}
 	}
+
+	if err = sessiontxn.OptimizeWithPlanAndThenWarmUp(c.Ctx, stmt.Plan); err != nil {
```

Comment:
Why move the `OptimizeWithPlanAndThenWarmUp` here from `ExecuteStmt`?

Reply:
If `IsPlanReplayerCaptureEnabled` is true, it will invoke `GetReadStmt` and activate the txn. Moving `OptimizeWithPlanAndThenWarmUp` here is to keep the pointget still be optimized before it.

Reply:
Better to add some comments about this constraint here or the possible implicit activation on the interface `GetStmtReadTS` and `GetStmtForUpdateTS`.

Reply:
Added some comments